hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -333,6 +338,7 @@ public boolean addSCM(AddSCMRequest request) throws IOException {
   @Override
   public List<DatanodeDetails> sortDatanodes(List<String> nodes,
       String clientMachine) throws IOException {
+    scm.checkAdminAccess(getRemoteUser(), false);
```

Comment:
sortDatanodes is kind of a read operation, it doesn't change the in memory or persist state of pipeline.

Reply:
Thanks @ChenSammi for review. I understood and handled the comment.